java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowFileReader.java

```diff
@@ -164,12 +146,54 @@ public boolean loadNextBatch() throws IOException {
       ArrowBlock block = footer.getRecordBatches().get(currentRecordBatch++);
       ArrowRecordBatch batch = readRecordBatch(in, block, allocator);
       loadRecordBatch(batch);
+      loadDictionaries();
```

Comment:
If loadDictionaries() throws an exception, does the batch leak?

Reply:
Yup, will fix.